lib/Home/data/datasource/weather/weather_data_source_local.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:warm_cloud/Home/data/datasource/weather/weather_data_source.dart';
+import 'package:warm_cloud/core/utility/shared_preferences_key.dart';
+
+class WeatherDataSourceLocal extends WeatherDataSource {
+  @override
+  Future<void> saveData() async {
+    final String jsonString = await rootBundle.loadString('assets/sample.json');
```

Comment:
asset path should come from constant file or static/const property of this class